### PR TITLE
Enhancement/auditcard

### PR DESCRIPTION
**Pull request by thorrester**
### Short Summary
Implements `AuditCard`

### Context
This is PR 1 of 2 for `opsml v1.5.0`. This PR adds `AuditCard` to the card registry. This stems from the AiSteerco work on auditability. In addition a sql migration script is created to add columns to Model and Data registries.

Overview:
- Adds template audit yaml questionnaire that is imported into the `AuditCard`
- DS has ability to list and answer questions as well as add comments 
- These methods are all used as part of new UI (will be in PR 2)
- New routes for listing teams and card names have been added (these will be used in PR 2)
- Remove some unused code + more tests

### Is this a Breaking Change?
No - adds new card to registry + sql migration script

### CODE REVIEW

File: opsml/app/routes/cards.py [hunk 1]
```diff
@@ -48,7 +50,62 @@ def check_uid(
     return UidExistsResponse(uid_exists=False)
 
 
-@router.post("/cards/version", response_model=Union[VersionResponse, UidExistsResponse], name="version")
+@router.get("/cards/teams", response_model=TeamsResponse, name="teams")
+def card_teams(
+    request: Request,
+    table_name: str,
```
Comment (damonallison) [inline opsml/app/routes/cards.py:56]: How about making "registry_type" a top level enum that we expose at the API layer? Exposing `table_name` isn't really the level that we want to expose, it's more like the registry name or type. like `Model` or `Data`. Internally it's a table, but externally it's a "registry".

File: opsml/app/routes/data.py [hunk 2]
```diff
@@ -75,7 +75,8 @@ def compare_data_profile(
 
     profiles = []
 
-    if payload.uids is not None:
+    # make mypy happy
```
Comment (damonallison) [inline opsml/app/routes/data.py:78]: Possible?

File: opsml/registry/cards/audit.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        return audit_sections
+
+
+class AuditQuestionTable:
+    """Helper class for creating a rich table to be used with an AuditCard"""
+
+    def __init__(self) -> None:
+        self.table = self.create_table()
+
+    def create_table(self):
```
Comment (damonallison) [inline opsml/registry/cards/audit.py:68]: add typing

File: opsml/registry/cards/audit.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+            Email to associate with the AuditCard
+        audit:
+            AuditSections object containing the audit questions and responses
+        audit_uri:
+            URI to the audit report
+        approved:
+            Whether the audit has been approved
+    """
+
+    audit: AuditSections = AuditSections()
```
Comment (damonallison) [inline opsml/registry/cards/audit.py:112]: Should these be at the instance, not class, level?

File: opsml/registry/sql/base/query_engine.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) Shipt, Inc.
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-
+# type: ignore
```
Comment (damonallison) [inline opsml/registry/sql/base/query_engine.py:4]: Can we remove this?

File: opsml/registry/sql/migration/alembic/versions/5b477729d615_add_auditcard_columns.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+def upgrade() -> None:
+    logger.info(f"Alembic revision: add auditcard_uid and uris cols - {revision}")
+
+    bind = op.get_context().bind
+    insp = sa.inspect(bind)
+    columns = insp.get_columns(RegistryTableNames.DATA.value)
+
+    for table_name in [RegistryTableNames.DATA.value, RegistryTableNames.MODEL.value]:
+        columns = insp.get_columns(table_name)
```
Comment (damonallison) [inline opsml/registry/sql/migration/alembic/versions/5b477729d615_add_auditcard_columns.py:31]: So audit can only be associated with data / model? Looks like the registry record allows associating audit w/ data / model / run. Perhaps run doesn't need a migration?

**Review by damonallison — DISMISSED** [review]

File: opsml/app/routes/cards.py (hunk 1 above)
Comment (thorrester) [reply opsml/app/routes/cards.py:56]: Good call. We should be keeping things consistent and simple. I'll update the logic

File: opsml/registry/cards/audit.py (hunk 4 above)
Comment (thorrester) [reply opsml/registry/cards/audit.py:112]: The ArtifactCard is a BaseModel, so these attributes should already be set at the instance level with defaults. Plus, having it this way makes it easier to load an auditcard from a yaml or csv definition as part of the UI

File: opsml/app/routes/data.py (hunk 2 above)
Comment (thorrester) [reply opsml/app/routes/data.py:78]: lol

File: opsml/registry/sql/migration/alembic/versions/5b477729d615_add_auditcard_columns.py (hunk 6 above)
Comment (thorrester) [reply opsml/registry/sql/migration/alembic/versions/5b477729d615_add_auditcard_columns.py]: good catch. added runcard table

**Pull request merged**